### PR TITLE
Invalidating token

### LINKED ISSUE #3923: Order-history request not fetching data
## Current behavior
I am authorized user. After some time my token is invalidated. I go to cart and refresh page. I see many refresh and order-history requests.  

image

I think, the problem is that order-history request doesnt use new token.,

## Expected behavior
I am authorized user.

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related issues
closes #3923

### Which environment this relates to
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance
- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/modules/user/index.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const UserModule: StorefrontModule = async function ({store}) {
     }
 
     if (
-      type.endsWith(types.USER_TOKEN_CHANGED)
+      type.endsWith(types.USER_TOKEN_CHANGED) && state.user.token
```
Comment (pkarw) [inline core/modules/user/index.ts]: are you sure about this second condition? I mean what about the log out/ clearing out the token on purpose? @andrzejewsky

Comment (andrzejewsky) [reply core/modules/user/index.ts]: I've checked it again - this condition is not even necessary

File: core/modules/user/index.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export const UserModule: StorefrontModule = async function ({store}) {
     if (
       type.endsWith(types.USER_TOKEN_CHANGED)
     ) {
```
Comment (pkarw) [inline core/modules/user/index.ts]: This should be probably removed?

Comment (andrzejewsky) [reply core/modules/user/index.ts]: removed

**Comment by andrzejewsky** [comment]
@gibkigonzo depends if that token is already set... it's super tricky, some time ago I spent more time on task.ts some of the logic there is not really obvious

**Comment by pkarw** [comment]
btw. you can easily test this feature by editing manually localStorage token (setting it to random value)

**Comment by pkarw** [comment]
@gibkigonzo good point - please do some testing if it solves the issue

**Comment by andrzejewsky** [comment]
@pkarw 

> btw. you can easily test this feature by editing manually localStorage token (setting it to random value) which will enforce token invalidation

this is exactly what i did

**Comment by andrzejewsky** [comment]
@gibkigonzo ok just checked, it actually make sense what you mentioned but.. it still triggers the same amount of calls like beforehand, also look at this line https://github.com/DivanteLtd/vue-storefront/blob/master/core/lib/sync/task.ts#L171 there is also `currentToken` that is passed to `_internalExecute` and when you use `userTokenInvalidated` instead of that... hmm it might break something...

**Pull request merged**